apps/src/lib/wallet/mod.rs

```diff
@@ -513,32 +635,110 @@ impl Wallet {
     }
 }
 
+/// Generates a random mnemonic of the given mnemonic type.
+fn generate_mnemonic_code(
+    mnemonic_type: MnemonicType,
+) -> Result<Mnemonic, GenRestoreKeyError> {
+    const BITS_PER_BYTE: usize = 8;
+
+    // generate random mnemonic
+    let entropy_size = mnemonic_type.entropy_bits() / BITS_PER_BYTE;
+    let mut bytes = vec![0u8; entropy_size];
+    rand::RngCore::fill_bytes(&mut rand::rngs::OsRng, &mut bytes);
+    let mnemonic = Mnemonic::from_entropy(&bytes, Language::English)
+        .expect("Mnemonic creation should not fail");
+
+    #[cfg(not(test))]
+    {
+        println!(
+            "Safely store your {} words mnemonic.",
+            mnemonic_type.word_count()
+        );
+        println!("{}", mnemonic.clone().into_phrase());
+    }
+
+    Ok(mnemonic)
+}
+
+fn get_secure_user_input<S>(request: S) -> std::io::Result<SecStr>
+where
+    S: std::fmt::Display,
+{
+    print!("{} ", request);
+    std::io::stdout().flush()?;
+
+    let mut response = String::new();
+    std::io::stdin().read_line(&mut response)?;
+    Ok(SecStr::from(response))
+}
+
+fn read_mnemonic_code() -> Result<Mnemonic, GenRestoreKeyError> {
+    let phrase = get_secure_user_input("Input mnemonic code: ")
+        .map_err(|_| GenRestoreKeyError::MnemonicInputError)?;
+
+    Mnemonic::from_phrase(
+        unsafe { std::str::from_utf8_unchecked(phrase.unsecure()) },
```

Comment:
The use of the `SecStr` is not playing nicely with the `Mnemoic` :/ Even though we could use the safe variant `from_utf8`, I'd be in favor of replacing it with zeroize because we're not using anything that's not already provided from zeroize and just use plain `zeroize::Zeroizing` over a String. This can then be easily passed into `Mnemoic::from_phrase` as it implements Deref

Reply:
SecStr is replaced with Zeroizing now.